Add searchable-snapshots-stats telemetry device

dliappis opened the pull request:
This commit adds a new telemetry device called
`searchable-snapshots-stats` which can be used to retrieve various
useful statistics about searchable snapshots while running a benchmark.

danielmitterdorfer reviewed (commented):
Thanks! I did an initial pass and left a few comments / suggestions.

dliappis commented on esrally/telemetry.py:
```diff
@@ -934,6 +936,188 @@ def record_stats_per_transform(self, transform_id, stats, prefix=""):
+                # allow collection, indices might be mounted later on
+                return
+        except elasticsearch.TransportError:
+            msg = f"A transport error occurred while collecting searchable snapshots stats on " \
```
Here we use msg in two places (contrary to above) i.e. to log the exception and raise a Rally Error, see below: https://github.com/elastic/rally/pull/1212/files#diff-fbfe3c50f180e0a3ac525aab1c0c2ef87ce11213e090850408a1e2383b813bdaR1077-R1078

We do this elsewhere too, of the top of my head I don't recall if it's sufficient to just `raise exceptions.RalyError(msg`) and trust that this results in a logged exception too.

dliappis commented:
@danielmitterdorfer that you for the quick review; I addressed your comments in https://github.com/elastic/rally/pull/1212/commits/e5310baf9f788f12fc785fa32d5dccf7d805da9f plus renamed a field as discussed offline in https://github.com/elastic/rally/pull/1212/commits/b201af5548ba3c5cbc5a4e0a9b2cb2d88d35426a. Once CI is green I'd appreciate a second look.

danielmitterdorfer commented on esrally/telemetry.py:
```diff
@@ -1067,13 +1067,15 @@ def record(self):
+                  f"[{self.cluster_name}]"
+
             self.logger.exception(msg)
             raise exceptions.RallyError(msg)
```
As this error will be caught elsewhere, does it make sense *not* to log this in the line above? If we want to keep the log, I wonder whether we should instead [raise from None](https://stefan.sofa-rockers.org/2020/10/28/raise-from/).

danielmitterdorfer reviewed (approved):
Thanks for iterating. I tested this as well and it looks good. I left one note about exception handling but no need for another review round in any case.
